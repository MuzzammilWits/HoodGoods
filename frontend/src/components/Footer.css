.footer {
    background-color: white;
    /* border-top: 1px solid #e0e0e0; */
  }
  
  .footer-main {
    padding: 3rem 0;
  }
  
  .footer-content {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 2rem;
  }
  
  .footer-logo {
    display: flex;
    align-items: center;
  }
  
  .footer-logo .logo {
    width: 100px;
    height: 100px;
  }
  
  .footer-menu h3,
  .footer-social h3 {
    font-size: 1.2rem;
    margin-bottom: 1.5rem;
    font-weight: 600;
  }
  
  .footer-menu ul {
    list-style: none;
    padding: 0;
    margin: 0;
  }
  
  .footer-menu li {
    margin-bottom: 0.75rem;
  }
  
  .footer-menu a {
    color: var(--text-color);
    transition: color 0.3s ease;
  }
  
  .footer-menu a:hover {
    color: var(--secondary-color);
  }
  
  .social-icons {
    display: flex;
    gap: 1rem;
  }
  
  .social-icon {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    background-color: var(--secondary-color);
    color: white;
    border-radius: 50%;
    transition: background-color 0.3s ease;
  }
  
  .social-icon:hover {
    background-color: #9a7763;
  }
  
  .social-icon svg {
    width: 20px;
    height: 20px;
  }
  
  .footer-bottom {
    padding: 1.5rem 0;
    border-top: 1px solid #e0e0e0;
  }
  
  .footer-bottom-content {
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .copyright {
    font-size: 0.9rem;
    color: #666;
  }
  
  .footer-links {
    display: flex;
    gap: 1.5rem;
  }
  
  .footer-links a {
    font-size: 0.9rem;
    color: #666;
    transition: color 0.3s ease;
  }
  
  .footer-links a:hover {
    color: var(--secondary-color);
  }
  
  @media (max-width: 768px) {
    .footer-content {
      grid-template-columns: 1fr;
      text-align: center;
    }
    
    .footer-logo {
      justify-content: center;
      margin-bottom: 1.5rem;
    }
    
    .social-icons {
      justify-content: center;
    }
    
    .footer-bottom-content {
      flex-direction: column;
      gap: 1rem;
    }
    
    .footer-links {
      justify-content: center;
    }
  }