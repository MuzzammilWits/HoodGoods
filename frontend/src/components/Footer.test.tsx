import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect, vi, beforeEach, afterEach } from 'vitest';
import { MemoryRouter } from 'react-router-dom'; // To wrap components using <Link>
import Footer from './Footer'; // Adjust the import path as needed

// --- Mocks ---

// Mock the logo import
vi.mock('./utils/ImageImports', () => ({
  logo: 'mock-logo.svg',
}));

// Mock window.scrollTo
const mockScrollTo = vi.fn();

describe('Footer Component', () => {
  let originalScrollTo: typeof window.scrollTo;

  beforeEach(() => {
    // Store original and assign mock
    originalScrollTo = window.scrollTo;
    window.scrollTo = mockScrollTo;

    render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
  });

  afterEach(() => {
    // Restore original
    window.scrollTo = originalScrollTo;
    vi.clearAllMocks();
  });

  // Test 1: Basic rendering of essential elements
  it('should render the footer logo, social icons, and copyright text', () => {
    // Check for the logo
    const logoImg = screen.getByAltText('Hood Goods');
    expect(logoImg).toBeInTheDocument();
    expect(logoImg).toHaveAttribute('src', 'mock-logo.svg');

    // Check for social icons (by aria-label)
    expect(screen.getByLabelText('Facebook')).toBeInTheDocument();
    expect(screen.getByLabelText('Twitter')).toBeInTheDocument();
    expect(screen.getByLabelText('Instagram')).toBeInTheDocument();

    // Check for copyright text
    expect(screen.getByText(/Copyright Â©HoodsGoods/i)).toBeInTheDocument();
  });

  // Test 2: "Back to Top" button functionality
  it('should render the "Back to Top" button and call window.scrollTo on click', () => {
    const backToTopButton = screen.getByRole('button', { name: /back to top/i });
    expect(backToTopButton).toBeInTheDocument();

    fireEvent.click(backToTopButton);
    expect(mockScrollTo).toHaveBeenCalledTimes(1);
    expect(mockScrollTo).toHaveBeenCalledWith({ top: 0, behavior: 'smooth' });
  });

  // Test 3: Social media links properties
  it('should render social media links with correct href, target, and rel attributes', () => {
    const socialLinks = [
      { label: 'Facebook', href: 'https://www.youtube.com/watch?v=xvFZjo5PgG0' },
      { label: 'Twitter', href: 'https://www.youtube.com/watch?v=xvFZjo5PgG0' },
      { label: 'Instagram', href: 'https://www.youtube.com/watch?v=xvFZjo5PgG0' },
    ];

    socialLinks.forEach(linkInfo => {
      const linkElement = screen.getByLabelText(linkInfo.label);
      expect(linkElement).toBeInTheDocument();
      expect(linkElement).toHaveAttribute('href', linkInfo.href);
      expect(linkElement).toHaveAttribute('target', '_blank');
      expect(linkElement).toHaveAttribute('rel', 'noopener noreferrer');
      // Check for SVG presence within the link
      expect(linkElement.querySelector('svg')).toBeInTheDocument();
    });
  });

  // Test 4: Footer navigation links
  it('should render footer navigation links for Terms & Conditions and Privacy Policy', () => {
    const termsLink = screen.getByRole('link', { name: /terms & conditions/i });
    expect(termsLink).toBeInTheDocument();
    expect(termsLink).toHaveAttribute('href', '/terms-and-conditions');

    const privacyLink = screen.getByRole('link', { name: /privacy policy/i });
    expect(privacyLink).toBeInTheDocument();
    expect(privacyLink).toHaveAttribute('href', '/privacy-policy');
  });

  // Test 5: Presence of the decorative SVG divider
  it('should render the decorative SVG section divider', () => {
    const dividerSvg = document.querySelector('.section-divider svg'); // More specific selector
    expect(dividerSvg).toBeInTheDocument();
    expect(dividerSvg?.querySelector('path[d*="M0,100 L 0,40"]')).toBeInTheDocument(); // Check for part of the path
  });

  // Test 6: Presence of the horizontal rule divider
  it('should render the horizontal rule divider', () => {
    const hrElement = screen.getByRole('separator', { hidden: true }); // hr has implicit role separator
    expect(hrElement).toBeInTheDocument();
    expect(hrElement).toHaveClass('footer-divider');
  });

  // Test 7: Snapshot test for overall structure
  it('should match the snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Footer />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
