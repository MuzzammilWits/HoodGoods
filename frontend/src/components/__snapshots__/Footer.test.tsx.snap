// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > should match the snapshot 1`] = `
<footer
  class="footer"
>
  <section
    class="footer-divider-section"
  >
    <figure
      class="section-divider"
      role="presentation"
    >
      <svg
        preserveAspectRatio="none"
        viewBox="0 0 100 100"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0,100 L 0,40 L 15,75 L 30,25 L 50,85 L 70,20 L 85,70 L 100,40 L 100,100 Z"
          fill="#432C53"
        />
      </svg>
    </figure>
  </section>
  <section
    class="footer-main"
  >
    <section
      class="footer-backtotop-area"
    >
      <button
        aria-label="Back to Top"
        class="back-to-top-btn"
      >
        Back to Top
      </button>
    </section>
    <div
      class="footer-content-row"
    >
      <section
        class="footer-logo-area"
      >
        <figure
          class="footer-logo"
        >
          <img
            alt="Hood Goods"
            class="logo"
            src="mock-logo.svg"
          />
        </figure>
      </section>
      <section
        class="footer-social-area"
      >
        <nav
          aria-label="Social Links"
          class="social-icons"
        >
          <a
            aria-label="Facebook"
            class="social-icon"
            href="https://www.youtube.com/watch?v=xvFZjo5PgG0"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12c0 4.84 3.44 8.87 8 9.8V15H8v-3h2V9.5C10 7.57 11.57 6 13.5 6H16v3h-2c-.55 0-1 .45-1 1v2h3v3h-3v6.95c5.05-.5 9-4.76 9-9.95z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="Twitter"
            class="social-icon"
            href="https://www.youtube.com/watch?v=xvFZjo5PgG0"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                fill="currentColor"
              />
            </svg>
          </a>
          <a
            aria-label="Instagram"
            class="social-icon"
            href="https://www.youtube.com/watch?v=xvFZjo5PgG0"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2c2.717 0 3.056.01 4.122.06 1.065.05 1.79.217 2.428.465.66.254 1.216.598 1.772 1.153a4.908 4.908 0 0 1 1.153 1.772c.247.637.415 1.363.465 2.428.047 1.066.06 1.405.06 4.122 0 2.717-.01 3.056-.06 4.122-.05 1.065-.218 1.79-.465 2.428a4.883 4.883 0 0 1-1.153 1.772 4.915 4.915 0 0 1-1.772 1.153c-.637.247-1.363.415-2.428.465-1.066.047-1.405.06-4.122.06-2.717 0-3.056-.01-4.122-.06-1.065-.05-1.79-.218-2.428-.465a4.89 4.89 0 0 1-1.772-1.153 4.904 4.904 0 0 1-1.153-1.772c-.248-.637-.415-1.363-.465-2.428C2.013 15.056 2 14.717 2 12c0-2.717.01-3.056.06-4.122.05-1.066.217-1.79.465-2.428a4.88 4.88 0 0 1 1.153-1.772A4.897 4.897 0 0 1 5.45 2.525c.638-.248 1.362-.415 2.428-.465C8.944 2.013 9.283 2 12 2zm0 1.802c-2.67 0-2.986.01-4.04.059-.976.045-1.505.207-1.858.344-.466.182-.8.398-1.15.748-.35.35-.566.683-.748 1.15-.137.353-.3.882-.344 1.857-.048 1.055-.058 1.37-.058 4.041 0 2.67.01 2.986.058 4.04.045.977.207 1.505.344 1.858.182.466.399.8.748 1.15.35.35.683.566 1.15.748.353.137.882.3 1.857.344 1.054.048 1.37.058 4.041.058 2.67 0 2.987-.01 4.04-.058.977-.045 1.505-.207 1.858-.344.466-.182-.8-.398 1.15-.748.35-.35.566-.683.748-1.15.137-.353-.3.882-.344 1.857.048-1.055.058-1.37.058-4.041 0-2.67-.01-2.986-.058-4.04-.045-.977-.207-1.505-.344-1.858a3.097 3.097 0 0 0-.748-1.15 3.098 3.098 0 0 0-1.15-.748c-.353-.137-.882-.3-1.857-.344-1.055-.048-1.37-.058-4.041-.058zm0 3.063a5.135 5.135 0 1 1 0 10.27 5.135 5.135 0 0 1 0-10.27zm0 8.468a3.333 3.333 0 1 0 0-6.666 3.333 3.333 0 0 0 0 6.666zm6.538-8.469a1.2 1.2 0 1 1-2.4 0 1.2 1.2 0 0 1 2.4 0z"
                fill="currentColor"
              />
            </svg>
          </a>
        </nav>
      </section>
      <nav
        aria-label="Footer Links"
        class="footer-links-area"
      >
        <li>
          <a
            class="hover:text-white"
            data-discover="true"
            href="/terms-and-conditions"
          >
            Terms & Conditions
          </a>
        </li>
        <li>
          <a
            class="hover:text-white"
            data-discover="true"
            href="/privacy-policy"
          >
            Privacy Policy
          </a>
        </li>
      </nav>
    </div>
    <hr
      aria-hidden="true"
      class="footer-divider"
    />
    <p
      class="copyright"
    >
      Copyright Â©HoodsGoods
    </p>
  </section>
</footer>
`;
